### PR TITLE
s4696561: Classifying Alzheimer's Disease using Vision Transformers

### PR DESCRIPTION
**Pull request by JoshuaWang8**
Hi COMP3710 Teaching Team,

This is my pull request for task 6 of the report. I have implemented a Vision Transformer capable of classifying whether a brain has Alzheimer's disease or not from MRI images, at a testing accuracy of around 65%. The files included in this pull request are:

- `README.md`: The README file for this project.
- `parameters.py`: Contains the parameters for the Vision Transformer model.
- `modules.py`: Contains the components of the model.
- `dataset.py`: Contains function for loading data.
- `train.py`: File for compiling, training and saving the Vision Transformer.
- `predict.py`: File for loading and predicting on the trained model.
- `utils.py`: Contains functions for visualizing data
- `resources/`: Directory containing resources used in `README.md`

Thanks,
Chiao-Hsi Joshua Wang (Student No. 46965611)

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
Excellent work.

Model design: Great
Data split: Correct
README: Clearly explained
Training process: Correct
Commit logs: Great.

**Comment by JoshuaWang8** [comment]
Thank you very much for the feedback. I wanted to point out that I have included a validation set during training, which I described under the  ["Training, Validation and Test Splits"](https://github.com/JoshuaWang8/PatternFlow/blob/topic-recognition/recognition/46965611-ADNI-Brain-Visual-Transformer/README.md#training-validation-and-test-splits) heading of my README. For each of the models, I have also plotted the accuracy and losses of both the training and validation sets, and the test results are mentioned for each model too.

**Comment by LinfengLiu98** [comment]
No worries, sorry I didn't see that part before. It won't affect your mark. Comments are fixed too. :)

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem (lower accuracy than expected) -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
